src/game.cpp

```diff
@@ -7141,6 +7141,16 @@ void game::smash()
             return; // don't smash terrain if we've smashed a corpse
         }
     }
+    bool ter_or_furn = m.has_flag_ter_or_furn( "ELECTRIFIED", smashx, smashy );
+    if (ter_or_furn) {
+            body_part bp = random_body_part();
```

Comment:
> Any other suggestions?

That's probably a bit complicated anyway. The zapback defense unconditionally damages the arms, that might be a good simple solution.
Another solution could be (not tested, identifiers are probably misspelled):
```C++
static const std::array<body_part, 6> suitable_bps = { bp_r_foot, bp_l_foot, ... };
const body_part target_bp = suitable_bps[rng(0, suitable_bps.size() - 1)];
```
But that gives each body part the same chance.